Admin/MenuBuilderInterface.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\FactoryInterface as MenuFactoryInterface;
+use Knp\Menu\ItemInterface;
+
+/**
+ * This interface can be implemented by admins that need to build menus.
+ *
+ * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ */
+interface MenuBuilderInterface
+{
+    /**
+     * @param MenuFactoryInterface $menuFactory
+     */
+    public function setMenuFactory(MenuFactoryInterface $menuFactory);
+
+    /**
+     * @return MenuFactoryInterface
+     */
+    public function getMenuFactory();
```

Comment:
> Should I really move them back? What exact methods would you move back and why?

Just talking about the `getMenuFactory` method. IMHO this method did only exist for BR reasons.


> Also, how do you deprecate an interface? And is removing methods from an interface a BC-break?

You're right. We must keep with the next major release, because we rely on it in many files.